api-client/src/endpoints/createShelterEndpoint.ts

```diff
@@ -6,14 +6,38 @@ import { JsonShelterType } from '../types'
 const SHELTER_URL = 'wohnraum.tuerantuer.org/wp-json/accommodations'
 
 export const SHELTER_ENDPOINT_NAME = 'shelter'
-type Params = { type: 'detail'; id: string; cityCode: string } | { type: 'list'; page: number; cityCode: string }
+type Params =
+  | { type: 'detail'; id: string; cityCode: string }
+  | { type: 'list'; page: number; cityCode: string; filter?: FilterProps }
+
+export type FilterProps = {
+  beds: string | null
+  pets: string | null
+}
+
+// Map for the filter keys for the endpoint
+const paramMap = new Map([
+  ['beds', 'min_beds'],
+  ['pets', 'pets']
+])
+
+const getFilterQueryParams = (params: FilterProps): string =>
+  Object.keys(params)
+    .map((key: string) => {
+      if (params[key as keyof FilterProps]) {
+        return `&${paramMap.get(key)}=${params[key as keyof FilterProps]}`
+      }
+      return ''
+    })
+    .join('')
 
 export default (): Endpoint<Params, ShelterModel[]> =>
   new EndpointBuilder<Params, ShelterModel[]>(SHELTER_ENDPOINT_NAME)
     .withParamsToUrlMapper(params => {
       const baseUrl = `https://${params.cityCode}.${SHELTER_URL}`
       if (params.type === 'list') {
-        return `${baseUrl}/list?page=${params.page}`
+        const filterParams = params.filter ? `${getFilterQueryParams(params.filter)}` : ''
+        return `${baseUrl}/list?page=${params.page}${filterParams}`
```

Comment:
Optional oc but seems a lot simpler to me.
```suggestion
        const { beds, pets } = params.filter
        let filterParams = beds ? `&min_beds=${beds}` : ''
        if (pets) {
          filterParams = `${filterParams}&pets=${pets}`
        }
        return `${baseUrl}/list?page=${params.page}${filterParams}`
```

Reply:
if we get more filters, you want to add every filter here? Don't think thats an improvement to be honest.

Reply:
yep, would do it that way, we will never have more then five filters. but your way is fine so just keep it.